runtime/compiler/aarch64/codegen/J9TreeEvaluator.cpp

```diff
@@ -101,6 +101,65 @@ J9::ARM64::TreeEvaluator::generateFillInDataBlockSequenceForUnresolvedField(TR::
    TR_ASSERT_FATAL(false, "This helper implements platform specific code for Fieldwatch, which is currently not supported on ARM64 platforms.\n");
    }
 
+TR::Register *
+J9::ARM64::TreeEvaluator::conditionalHelperEvaluator(TR::Node *node, TR::CodeGenerator *cg)
+   {
+
+   TR::RegisterDependencyConditions *conditions;
+   TR::Register *jumpReg, *valReg;
+   TR::Node *testNode = node->getFirstChild();
+   TR::Node *callNode = node->getSecondChild();
+   TR::Node *firstChild = testNode->getFirstChild();
+   TR::Node *secondChild = testNode->getSecondChild();
+   int32_t value, i, numArgs = callNode->getNumChildren();
+
+   conditions = new (cg->trHeapMemory()) TR::RegisterDependencyConditions(3, 3, cg->trMemory());
+
+   TR_ASSERT(numArgs <= 2, "Unexpected number of arguments for helper.");
+
+   // Helper arguments are in reversed order of the private linkage
+   // Argument registers are not needed to be split since the helper will
+   // preserve all of them.
+   int32_t iArgIndex = 0, fArgIndex = 0;
+   TR::Linkage *linkage = cg->createLinkage(TR_Private);
+   for (i = numArgs - 1; i >= 0; i--)
+      {
+      TR::Register *argReg = cg->evaluate(callNode->getChild(i));
+      TR::addDependency(conditions, argReg, (argReg->getKind() == TR_GPR) ? // Didn't consider Long here
+            linkage->getProperties().getIntegerArgumentRegister(iArgIndex++) : linkage->getProperties().getFloatArgumentRegister(fArgIndex++), argReg->getKind(), cg);
+      }
+
+   jumpReg = cg->evaluate(firstChild);
+   TR::addDependency(conditions, jumpReg, TR::RealRegister::x0, TR_GPR, cg);
+   bool is64Bit = node->getSecondChild()->getType().isInt64();
+   value = is64Bit ? secondChild->getLongInt() : secondChild->getInt();
+   if (secondChild->getOpCode().isLoadConst() && constantIsUnsignedImm12(value))
+      {
+      generateCompareImmInstruction(cg, testNode, jumpReg, value);
+      }
+   else
+      {
+      valReg = cg->evaluate(secondChild);
+      generateCompareInstruction(cg, testNode, jumpReg, valReg);
+      }
+   TR::Instruction *gcPoint;
+   TR::LabelSymbol *snippetLabel = generateLabelSymbol(cg);
+   cg->addSnippet(new (cg->trHeapMemory()) TR::ARM64HelperCallSnippet(cg, node, snippetLabel, node->getSymbolReference()));
+   if(testNode->getOpCodeValue() == TR::icmpeq)
+      gcPoint = generateConditionalBranchInstruction(cg, TR::InstOpCode::b_cond, node, snippetLabel, TR::CC_EQ, conditions);
+   else
+      gcPoint = generateConditionalBranchInstruction(cg, TR::InstOpCode::b_cond, node, snippetLabel, TR::CC_NE, conditions);
```

Comment:
The two sides of the branch are identical except for the choice of condition code.  To save some code space, can you simplify this to:
```
TR::ARM64ConditionCode cc = (testNode->getOpCodeValue() == TR::icmpeq) ? TR::CC_EQ : TR::CC_NE;
gcPoint = generateConditionalBranchInstruction(cg, TR::InstOpCode::b_cond, node, snippetLabel, cc, conditions);
```